src/widgets/dialogs/SelectChannelDialog.cpp

```diff
@@ -302,90 +278,109 @@ bool SelectChannelDialog::hasSeletedChannel() const
+                    ui.automod->setFocus();
+                    return true;
+                }
+
+                if (widget == ui.whispers)
+                {
+                    ui.channel->setFocus();
+                    return true;
+                }
+
+                auto *previousInFocusChain = widget->previousInFocusChain();
+                if (previousInFocusChain->focusPolicy() ==
+                    Qt::FocusPolicy::NoFocus)
+                {
+                    // Make sure we're not selecting one of the labels
+                    previousInFocusChain =
+                        previousInFocusChain->previousInFocusChain();
+                }
+                previousInFocusChain->setFocus();
                 return true;
             }
 
-            widget->nextInFocusChain()->setFocus();
-            return true;
-        }
-        else if (((event_key->key() == Qt::Key_Tab ||
-                   event_key->key() == Qt::Key_Backtab) &&
-                  event_key->modifiers() == Qt::ShiftModifier) ||
-                 ((event_key->key() == Qt::Key_Up) &&
-                  event_key->modifiers() == Qt::NoModifier))
-        {
-            // Shift+Tab has been pressed, focus previous entry in list
-
-            if (widget == this->dialog->ui_.twitch.channelName)
+            if (keyEvent == QKeySequence::DeleteStartOfWord &&
+                ui.channelName->selectionLength() > 0)
             {
-                // Special case for when current selection is the "Channel" entry's edit box since the Edit box actually has the focus
-                this->dialog->ui_.twitch.automod->setFocus();
+                ui.channelName->backspace();
                 return true;
             }
 
-            widget->previousInFocusChain()->setFocus();
-            return true;
-        }
-        else if (event_key == QKeySequence::DeleteStartOfWord &&
-                 this->dialog->ui_.twitch.channelName->selectionLength() > 0)
-        {
-            this->dialog->ui_.twitch.channelName->backspace();
-            return true;
-        }
-        else
-        {
             return false;
         }
-    }
-    else if (event->type() == QEvent::KeyRelease)
-    {
-        QKeyEvent *event_key = static_cast<QKeyEvent *>(event);
-        if ((event_key->key() == Qt::Key_Backtab ||
-             event_key->key() == Qt::Key_Down) &&
-            event_key->modifiers() == Qt::NoModifier)
-        {
-            return true;
+        break;
+
+        case QEvent::KeyRelease: {
+            auto *keyEvent = dynamic_cast<QKeyEvent *>(event);
+            assert(keyEvent);
+
+            if ((keyEvent->key() == Qt::Key_Backtab ||
+                 keyEvent->key() == Qt::Key_Down) &&
+                keyEvent->modifiers() == Qt::NoModifier)
+            {
+                return true;
+            }
         }
+        break;
     }
```

Comment:
Missing a default case.

Reply:
fixed in [`34ccfad` (#6081)](https://github.com/Chatterino/chatterino2/pull/6081/commits/34ccfad5f9ff5a3ffa578e8bf6e843c96ebefe36)